src/components/PlotWrapper.tsx

```diff
@@ -66,5 +69,20 @@ export default function PlotWrapper(inputProps: PlotWrapperProps): ReactElement
     // return () => window.removeEventListener("resize", updatePlotSize);
   }, [plot, plotDivRef.current]);
 
+  useEffect(() => {
+    const onClickPlot = (eventData: Plotly.PlotMouseEvent): void => {
+      if (eventData.points.length === 0) {
+        return;
+      }
+      const time = eventData.points[0].x as number;
+      props.setFrame(time);
+    };
+
+    (plotDivRef.current as PlotlyHTMLElement | null)?.on("plotly_click", onClickPlot);
+    return () => {
+      (plotDivRef.current as PlotlyHTMLElement | null)?.removeAllListeners("plotly_click");
```

Comment:
would using a local variable for clarity make the code more readable and less error prone? e.g. `const plotDiv = plotDivRef.current as PlotlyHTMLElement | null` - if `plotDiv` is defined, we can then setup the listener

Reply:
Yeah! Made this change, though note that I dereferenced the `plotDiv` again in the returned callback so I make sure I always have the freshest value.